cmake/helpers/GdalGenerateConfig.cmake

```diff
@@ -1,3 +1,23 @@
+if (NOT WINDOWS OR MINGW)
+    set(split_libs_default ON)
+else ()
+    set(split_libs_default OFF)
+endif ()
+option(GDAL_SPLIT_EXPORTED_LIBS "Split library path and name on export" ${split_libs_default})
```

Comment:
I didn't apply this option to the gdal lib itself: 
It would require a little bit more code to do correctly (static/shared/import...). But it would make sense with regard to consistency.